tensorflow/lite/kernels/neg_test.cc

```diff
@@ -75,5 +103,61 @@ TEST(NegOpModel, NegInt64) {
   EXPECT_THAT(m.GetOutput<int64_t>(), ElementsAreArray({2, 1, 0, -1, -2, -3}));
 }
 
+class NegOpQuantizedModel : public NegOpModel {
+ public:
+  NegOpQuantizedModel(TensorData input, TensorData output)
+      : NegOpModel(SymmetricInt16Scaling(std::move(input)),
+                   SymmetricInt16Scaling(std::move(output))) {}
+
+  template <typename integer_dtype>
+  std::vector<float> GetDequantizedOutput() {
+    return Dequantize<integer_dtype>(ExtractVector<integer_dtype>(output_),
+                                     GetScale(output_), GetZeroPoint(output_));
+  }
+
+ private:
+  TensorData SymmetricInt16Scaling(TensorData tensor) {
+    if (tensor.type == TensorType_INT16) {
+      CHECK_EQ(std::abs(tensor.min), tensor.max);
+      tensor.scale = tensor.max / std::numeric_limits<int16_t>::max();
+      tensor.zero_point = 0;
+      tensor.min = 0;
+      tensor.max = 0;
+    }
+    return tensor;
+  }
+};
+
+template <typename T>
+float GetTolerance(float min, float max) {
+  float kQuantizedStep =
+      2.0 * (max - min) /
+      (std::numeric_limits<T>::max() - std::numeric_limits<T>::min());
+  return kQuantizedStep;
+}
+
+template <TensorType tensor_type, typename integer_dtype>
+void QuantizedTests() {
+  float kQuantizedTolerance = GetTolerance<integer_dtype>(-128.0, 128.0);
+  std::vector<float> input = {-128.0f, -9, 0, 8, 127};
```

Comment:
const

Reply:
Addressed the requested change in [#ee05fdd](https://github.com/MCW-Dev/tensorflow/pull/49/commits/ee05fdd3332fb12a710b5e3d1cdab0b3b3e390ca)